athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java

```diff
@@ -0,0 +1,169 @@
+import com.amazonaws.athena.connector.lambda.metadata.GetTableLayoutRequest;
+import com.amazonaws.connectors.athena.jdbc.connection.DatabaseConnectionConfig;
+import com.amazonaws.connectors.athena.jdbc.connection.GenericJdbcConnectionFactory;
+import com.amazonaws.connectors.athena.jdbc.connection.JdbcConnectionFactory;
+import com.amazonaws.connectors.athena.jdbc.manager.JDBCUtil;
+import com.amazonaws.connectors.athena.jdbc.manager.JdbcMetadataHandler;
+import com.amazonaws.services.athena.AmazonAthena;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Handles metadata for Snowflake. User must have access to `schemata`, `tables`, `columns`, `partitions` tables in
+ * information_schema.
+ */
+public class SnowflakeMetadataHandler
+        extends JdbcMetadataHandler
+{
+    static final Map<String, String> JDBC_PROPERTIES = ImmutableMap.of("databaseTerm", "SCHEMA");
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeMetadataHandler.class);
+
+    /**
+     * Instantiates handler to be used by Lambda function directly.
+     *
+     * Recommend using {@link com.amazonaws.connectors.athena.jdbc.MultiplexingJdbcCompositeHandler} instead.
+     */
+    public SnowflakeMetadataHandler()
+    {
+        this(JDBCUtil.getSingleDatabaseConfigFromEnv(JdbcConnectionFactory.DatabaseEngine.SNOWFLAKE));
+    }
+
+    public SnowflakeMetadataHandler(final DatabaseConnectionConfig databaseConnectionConfig)
+    {
+        super(databaseConnectionConfig, new GenericJdbcConnectionFactory(databaseConnectionConfig, JDBC_PROPERTIES));
+    }
+
+    @VisibleForTesting
+    protected SnowflakeMetadataHandler(final DatabaseConnectionConfig databaseConnectionConfig, final AWSSecretsManager secretsManager,
+            final AmazonAthena athena, final JdbcConnectionFactory jdbcConnectionFactory)
+    {
+        super(databaseConnectionConfig, secretsManager, athena, jdbcConnectionFactory);
+    }
+
+    @Override
+    protected String escapeNamePattern(final String name, final String escape)
+    {
+        return super.escapeNamePattern(name, escape).toUpperCase();
+    }
+
+    @Override
+    protected ResultSet getColumns(final String catalogName, final TableName tableHandle, final DatabaseMetaData metadata)
+            throws SQLException
+    {
+        String escape = metadata.getSearchStringEscape();
+        LOGGER.warn(
+          "[SNOWFLAKE DEBUG] catalogName: {}, SchemaName: {}, TableName: {}",
```

Comment:
Logging to warm but text says debug

Reply:
Yeah the PR is not ready yet, shouldn't have pushed my debug lol